src/Sulu/Bundle/MediaBundle/Resources/js/containers/SingleMediaSelection/SingleMediaSelection.js

```diff
@@ -0,0 +1,139 @@
+    disabled: boolean,
+    locale: IObservableValue<string>,
+    onChange: (selectedIds: Value) => void,
+    value: Value,
+|}
+
+@observer
+export default class SingleMediaSelection extends React.Component<Props> {
+    static defaultProps = {
+        disabled: false,
+        value: { id: undefined },
+    };
+
+    singleMediaSelectionStore: SingleMediaSelectionStore;
+    changeDisposer: () => void;
+    changeAutorunInitialized: boolean = false;
+
+    @observable overlayOpen: boolean = false;
+
+    constructor(props: Props) {
+        super(props);
+
+        const {onChange, locale, value} = this.props;
+
+        this.singleMediaSelectionStore = new SingleMediaSelectionStore(value.id, locale);
+        this.changeDisposer = autorun(() => {
+            const {value} = this.props;
+            const loadedMediaId = this.singleMediaSelectionStore.selectedMediaId;
+
+            if (!this.changeAutorunInitialized) {
+                this.changeAutorunInitialized = true;
+                return;
+            }
+
+            if (value.id === loadedMediaId) {
+                return;
+            }
+
+            onChange({ id: loadedMediaId });
+        });
+    }
+
+    componentDidUpdate() {
+        const {
+            locale,
+            value,
+        } = this.props;
+
+        const newSelectedId = toJS(value.id);
+        const loadedSelectedId = toJS(this.singleMediaSelectionStore.selectedMediaId);
+
+        if (loadedSelectedId !== newSelectedId) {
+            this.singleMediaSelectionStore.loadSelectedMedia(newSelectedId, locale);
+        }
+    }
+
+    componentWillUnmount() {
+        this.changeDisposer();
+    }
+
+    @action openOverlay() {
+        this.overlayOpen = true;
+    }
+
+    @action closeOverlay() {
+        this.overlayOpen = false;
+    }
+
+    handleRemove = () => {
+        this.singleMediaSelectionStore.clear();
+    };
+
+    handleOverlayOpen = () => {
+        this.openOverlay();
+    };
+
+    handleOverlayClose = () => {
+        this.closeOverlay();
+    };
+
+    handleOverlayConfirm = (selectedMedia: Object) => {
+        this.singleMediaSelectionStore.set(selectedMedia);
+        this.closeOverlay();
+    };
+
+    render() {
+        const {
+            disabled,
+            locale,
+        } = this.props;
+        const {
+            selectedMedia,
+            selectedMediaId,
+        } = this.singleMediaSelectionStore;
+
+        return (
+            <Fragment>
+                <SingleItemSelection
+                    disabled={disabled}
+                    emptyText={translate('sulu_media.select_media_singular')}
+                    leftButton={{
+                        icon: 'su-image',
+                        onClick: this.handleOverlayOpen,
+                    }}
+                    onRemove={this.singleMediaSelectionStore.selectedMedia ? this.handleRemove : undefined}
+                >
+                    {selectedMedia &&
+                    <MediaSelectionItem
```

Comment:
Usually we are adding padding to this component, so that you see it belongs to the above `&&`.